src/crate/testing/layer.py

```diff
@@ -93,21 +99,37 @@ def create_settings(self,
             settings.update(further_settings)
         return settings
 
-    def stop(self):
-        # override because if we use proc.kill the terminal gets poisioned
-        self.process.send_signal(signal.SIGINT)
-        self.process.wait()
+    def wdPath(self):
+        return self._wd
+
+    @property
+    def crate_servers(self):
+        return [self.http_url]
+
+    def setUp(self):
+        self.start()
 
     def start(self):
-        wd = self.wdPath()
-        self.start_cmd = self.start_cmd + ('-Des.path.data="%s"' % wd,)
-        super(CrateLayer, self).start()
-        self.wait_for_start()
-        self.wait_for_master()
+        if os.path.exists(self._wd):
+            shutil.rmtree(self._wd)
+
+        self.process = subprocess.Popen(self.start_cmd)
+        returncode = self.process.poll()
+        if returncode is not None:
+            raise SystemError('Failed to start server rc={0} cmd={1}'.format(
+                returncode, self.start_cmd))
+        self._wait_for_start()
+        self._wait_for_master()
+
+    def stop(self):
+        self.process.kill()
```

Comment:
didn't see any terminal poisoning locally or on travis. So I assumed that it is no longer true